### ISSUE TITLE
Backend Dev: Funder users can configure their own email alerts

### ISSUE THREAD

**Issue by jel535s** [MidwestAccessCoalition/lockbox_rails#586]
## Context
Currently, all Support Request actions trigger an email to all Funder users from a certain funder, regardless of who created the Support Request. We had a user provide direct feedback that they would like to be able to configure this to only see Support Requests they created.

The design here is still pending but the backend work can start now! The frontend work will be handled on TICKET and you'll want to pair to hook up the frontend dev work with the backend work.

## Process
Based on the designs from #585, on the Edit Profile page, add a new field that will allow Funder users to adjust their email notifications.

Potential wording-
Email Notifications Preference:
I want emails about every Support Request
or
I only want emails about Support Requests I created.

## Acceptance Criteria/Definition of Done
Funder users (currently just MAC users) are able to configure which email alerts they get, by using the new field in their Edit Profile page. Effective as soon as their preference is saved, the email alerts are updated to only send what they request.

## Resources
Pending from #585

**Comment by misternu** [comment]
@jel535s @jasonwynkoop86 Currently reviewing what's done of #589, and poking around these other changes and discussions. We do not currently store the address of a user, and I'm not sure there's a compelling reason to do so.

It would also be useful for devs to know where/if a user is redirected after editing the password or user information. Redirects to the main page made sense when we were editing less information, and this may no longer make sense.

If saving/submitting email notifications is going to be attached to the user info submit button, are we sure that the user will know that their email preference hasn't been saved?

**Comment by jasonwynkoop86** [comment]
When things are updated currently it returns to the main page with a green attention box. 
![image](https://user-images.githubusercontent.com/82782739/120871339-f3a2d800-c560-11eb-80a0-58c02e20e8cc.png)
Can we refresh the screen on update to show an attention box like the image above for each parameter that has been updated?
And another update for updating time zone, name, and email preferences. 

We don't need to have an address or change address form, but some of the partners have addresses tied to the accounts and if they were to be edited it should be through this universal place. Keeping all profile info that is editable in one place.

**Comment by nsiwnf** [comment]
@jel535s @BintLopez  Admins/MAC users can select to either get notifications from all user actions or just other admin actions?
We have note notification logic as 
>MAC users only get emailed about manual notes
MAC users don't get emailed about notes they wrote
Partner users get emailed about all notes

And insufficient funds/low balance as 
>All admins and lockbox email

**Comment by BintLopez** [comment]
Since the context in which this arose was for support request activity, I think we should keep the scope to that.

For all other admin activity (like balance alerts), I'd say we leave this alone for the time being. These alerts are not scoped to a specific support request and so I don't think should be suppressed or restricted.

HOWEVER -- for notes -- since notes are scoped to a support request, I'd say they fall in scope for the requested changes here. I think it would make sense to do the following...
- If an "admin" user were to set their notifications to ONLY those support requests that are theirs (aka support request is associated with that user), then it _would_ make sense for them to ONLY receive alerts for notes associated to their support requests and not all other notes.

**Comment by BintLopez** [comment]
@jel535s @jasonwynkoop86 -- I think we need to make sure to keep a user's account settings separate from the organization's account settings. Since the address is a field on the organization/lockboxpartner level vs an individual user, we should keep where we're managing/editing these things separately.

**Comment by jasonwynkoop86** [comment]
> @jel535s @jasonwynkoop86 -- I think we need to make sure to keep a user's account settings separate from the organization's account settings. Since the address is a field on the organization/lockboxpartner level vs an individual user, we should keep where we're managing/editing these things separately.

Good Point! i have mostly been interacting with the demo through the admin side. Items for user side will now be

Password Change
Name
Email Preferences
Time Zone

![image](https://user-images.githubusercontent.com/82782739/121817085-cdf28e80-cc44-11eb-8932-ae745b77731c.png)

**Comment by juliaBP** [comment]
@BintLopez how can I assign this to myself?
